kernel/src/process.rs

```diff
@@ -428,6 +428,15 @@ impl TbfHeader {
             _ => (0, 0),
         }
     }
+
+    /// Get the total size of writable flash regions.
+    fn get_total_writeable_flash_size(&self) -> u32 {
```

Comment:
This function is fine, but it will probably give people the impression that the writeable flash regions are contiguous, which need not be the case.

Reply:
This is definitely based on an assumption that all writeable flash regions are contiguous and between the TBFHeader and application header. If I'm not allowed to make those assumptions though, I'm not sure how to fix this problem.

crt0.c isn't given any information about the existence (or not) of writeable flash regions. And the current implementation of writeable flash regions places them in Flash immediately after the TBFHeader but before the application header. But, crt0.c assumes that the very first thing in the text segment is its own application header, screwing everything up.

Maybe this could be fixed by placing the .app_state section after the application header but before .start?

Reply:
Why does the app state affect crt0 at all? The app/crt0 header should be relative to _start (or its own text section start) which it should know. Having app state after the TBFheader and before _text shouldn't be any different than just having a much larger TBFheader.

And apps can get info on writeable flash regions through the memop syscall.